hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/jmx/TestJMXJsonServlet.java

```diff
@@ -62,10 +62,15 @@ public static void assertReFind(String re, String value) {
     result = readOutput(new URL(baseUrl, "/jmx?qry=java.lang:type=Memory"));
     assertReFind("\"name\"\\s*:\\s*\"java.lang:type=Memory\"", result);
     assertReFind("\"modelerType\"", result);
-    
+
+    System.setProperty("THE_TEST_OF_THE_NAN_VALUES", String.valueOf(Float.NaN));
     result = readOutput(new URL(baseUrl, "/jmx"));
     assertReFind("\"name\"\\s*:\\s*\"java.lang:type=Memory\"", result);
-    
+    assertReFind(
```

Comment:
How about add another case to verify the result when enable JMX_NAN_FILTER?

Reply:
Hi @Hexiaoqiao !

Thanks for the review!
Yes, I think it would be good to have a test like that, but that change just won't be nice I am afraid.
The problem is that there is no nice way to connect the **HttpServerFunctionalTest#createTestServer(Configuration conf)** to the **JMXJsonServlet**, cause in the **HttpServlet2#addDefaultServlets** we just provide class reference.
A possible solution is to create another JMX servlet class to do the trick.
I would prefer the previous code, cause that was less ugly, but I am open to keeping this one, to have the the test.

Reply:
Got it. Make sense to me.